### ISSUE TITLE
Parse Enum values as literals

### ISSUE THREAD

**Issue by notaphplover** [bcherny/json-schema-to-typescript#508]
### Description

Enum values should be parsed as literals as the spec states.

#### Expected behavior:

The following JSONSchema:

```JSON
{
  "title": "Enum with JSON literal",
  "enum": [ { "type": "string" } ]
}
```

should be transpiled into the following typescript model:

```typescript
export type EnumWithJSONLiteral = {type: "string"};
```

### Current behavior:

The following JSONSchema:

```JSON
{
  "title": "Enum with JSON literal",
  "enum": [ { "type": "string" } ]
}
```

is transpiled into the following typescript model:

```typescript
export type EnumWithJSONLiteral = string;
```

### Context

Consider the specs as references:

- https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-5.20
- https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-01#section-6.23
- https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-validation-00#section-6.1.2
- https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-validation-01#section-6.1.2
- https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-validation-02#section-6.1.2
- https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-validation-01#enum

**Comment by notaphplover** [comment]
@bcherny I just created #509 to fix this issue :)

**Comment by bcherny** [comment]
Published json-schema-to-typescript-12.0.0
